pdp/network_tree.go

```diff
@@ -0,0 +1,146 @@
+package pdp
+
+import (
+	"encoding/binary"
+	"fmt"
+	"github.com/miekg/bitradix"
+	"net"
+)
+
+type SetOfNetworks struct {
+	root *bitradix.Radix32
+}
+
+type NetworkLeafItem struct {
+	Network *net.IPNet
+	Leaf    interface{}
+}
+
+var (
+	masks = []net.IPMask{
+		net.CIDRMask(0x00, 32),
+		net.CIDRMask(0x01, 32),
+		net.CIDRMask(0x02, 32),
+		net.CIDRMask(0x03, 32),
+		net.CIDRMask(0x04, 32),
+		net.CIDRMask(0x05, 32),
+		net.CIDRMask(0x06, 32),
+		net.CIDRMask(0x07, 32),
+		net.CIDRMask(0x08, 32),
+		net.CIDRMask(0x09, 32),
+		net.CIDRMask(0x0a, 32),
+		net.CIDRMask(0x0b, 32),
+		net.CIDRMask(0x0c, 32),
+		net.CIDRMask(0x0d, 32),
+		net.CIDRMask(0x0e, 32),
+		net.CIDRMask(0x0f, 32),
+		net.CIDRMask(0x10, 32),
+		net.CIDRMask(0x11, 32),
+		net.CIDRMask(0x12, 32),
+		net.CIDRMask(0x13, 32),
+		net.CIDRMask(0x14, 32),
+		net.CIDRMask(0x15, 32),
+		net.CIDRMask(0x16, 32),
+		net.CIDRMask(0x17, 32),
+		net.CIDRMask(0x18, 32),
+		net.CIDRMask(0x19, 32),
+		net.CIDRMask(0x1a, 32),
+		net.CIDRMask(0x1b, 32),
+		net.CIDRMask(0x1c, 32),
+		net.CIDRMask(0x1d, 32),
+		net.CIDRMask(0x1e, 32),
+		net.CIDRMask(0x1f, 32),
+		net.CIDRMask(0x20, 32)}
+
+	ErrorIPv6NotImplemented = fmt.Errorf("IPv6 and mixed network sets haven't been implemented yet")
```

Comment:
The issue here that the bitradix package supports only 32 and 64 bits keys for IPv6 we need 128 bits. It can be implemented using nested 64 trees and requires to pay attention on different corner cases which results from cross tree address masks. I'd say it may take whole day so prefer to do it later. We can add the task to next sprint.